typescript/introduction/tsc-data-types-i/tsc-array-tuple.md

```diff
@@ -0,0 +1,119 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Array & Tuple
+
+---
+## Content
+
+### Array
+
+An array represents a collection of values. In TypeScript, there are two ways in which you can define an array:
+
+```ts
+// by putting the [] after the type
+// here we define an array of numbers
+let numberArray: number[] = [1, 3, 5, 7];
+
+// or by wraping your type in the Array type
+// here we define an array of strings
+let stringArray: Array<string> = [
+  "one",
+  "three",
+  "five",
+  "seven"
+];
+```
+
+Note that arrays can only hold values of the type that you specify.
+
+If you want to store more than one type, you can use a union:
+
+```ts
+// array of strings and numbers
+let mixedArray: Array<string|number>[] = [
+  1, 
+  'one', 
+  3, 
+  5, 
+  'five'
+];
+```
+
+### Tuple
+
+Tuples are like an array, but they can only take in a fixed number of elements with known types:
+
+```ts
+// myTyple has a number at the first position
+// and a boolean at the second position
+let myTuple: [number, boolean] = [
+  1, 
+  true
+];
+```
+
+Note that for tuples, the order of elements matters. 
+
+If you define its type as `[number, boolean]`, then the tuple must contain exactly two elements and be in the correct order. 
+
+Here is an example that would throw an error:
```

Comment:
let's also show the flipped, valid example

Reply:
We show it just above. Are you saying that we should repeat it once more or that we should somehow rephrase this?